toolchain/driver/clang_runner.cpp

```diff
@@ -146,7 +146,28 @@ auto ClangRunner::Run(llvm::ArrayRef<llvm::StringRef> args) -> bool {
   // Note the subprocessing will effectively call `clang -cc1`, which turns into
   // `carbon-busybox clang -cc1`, which results in an equivalent `clang_main`
   // call.
-  auto cc1_main = [](llvm::SmallVectorImpl<const char*>& cc1_args) -> int {
+  //
+  // Also note that we only do `-disable-free` filtering in the in-process
+  // execution here, as subprocesses leaking memory won't impact this process.
+  auto cc1_main = [enable_leaking = enable_leaking_](
```

Comment:
Why the explicit rename instead of something like:

```suggestion
  auto cc1_main = [enable_leaking_](
```

Reply:
`error: class member 'enable_leaking_' cannot appear in capture list as it is not a variable`

And more generally, it seems good to make it explicit that this lambda doesn't depend on any state outside of itself.